packages/esm-patient-chart-app/src/visit/visit-history-table/visit-history-table.component.tsx

```diff
@@ -87,13 +104,23 @@ const VisitHistoryTable: React.FC<VisitHistoryTableProps> = ({ patientUuid }) =>
                   </TableRow>
                 </TableHead>
                 <TableBody>
-                  {rows.map((row) => {
+                  {rows.map((row, i) => {
+                    const visit = visits[i];
                     return (
-                      <TableRow key={row.id} {...getRowProps({ row })}>
-                        {row.cells.map((cell) => {
-                          return <TableCell key={cell.id}>{cell?.value}</TableCell>;
-                        })}
-                      </TableRow>
+                      <React.Fragment key={row.id}>
+                        <TableExpandRow {...getRowProps({ row })}>
+                          {row.cells.map((cell) => {
+                            return <TableCell key={cell.id}>{cell?.value}</TableCell>;
+                          })}
+                        </TableExpandRow>
+                        <TableExpandedRow
+                          {...getExpandedRowProps({ row })}
+                          className={styles.expandedRow}
+                          colSpan={headers.length + 2}
+                        >
```

Comment:
To fix the double vertical borders on the table rows, you'll need to follow [this pattern](https://github.com/openmrs/openmrs-esm-patient-chart/blob/main/packages/esm-patient-notes-app/src/notes/paginated-notes.component.tsx#L124):

```tsx
row.isExpanded ? (
  <TableExpandedRow
    {...getExpandedRowProps({ row })}
    className={styles.expandedRow}
    colSpan={headers.length + 2}
  >
    <VisitSummary visit={visit} patientUuid={patientUuid} mutateVisit={mutate} />
  </TableExpandedRow>
  ) : (
    <TableExpandedRow className={styles.hiddenRow} colSpan={headers.length + 2} />
  )}
```

where `.hiddenRow` is:

```scss
.hiddenRow {
  display: none;
}
```